authentication/context_authenticator.go

```diff
@@ -0,0 +1,62 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+const (
+	// Scheme is the Authorization header scheme
+	Scheme = "bearer"
+
+	// HeaderKey is the key used for the authentication information in headers
+	HeaderKey = "Authorization"
+)
+
+var errInvalidHeader = errors.New("invalid authorization header")
+
+// NewContextAuthenticator creates a function that stores authentication information in context
+func NewContextAuthenticator(secret []byte) func(ctx context.Context, authHeader string) (context.Context, error) {
```

Comment:
Id rather call this NewMiddleware. maybe the other one should be on a http pkg or something like that.